lib/src/driver.dart

```diff
@@ -353,13 +353,14 @@ class Driver {
       });
     }
 
-    AnalysisEngine.instance.logger = new StdLogger(options.log);
+    if (options.log) {
+      AnalysisEngine.instance.logger = new StdLogger();
+    }
 
     // Set context options.
     AnalysisOptionsImpl contextOptions = new AnalysisOptionsImpl();
     contextOptions.cacheSize = _maxCacheSize;
     contextOptions.hint = !options.disableHints;
-    contextOptions.enableNullAwareOperators = options.enableNullAwareOperators;
```

Comment:
This is a significant enough change that I'd prefer for it to be in its own CL.  Also, we can't do this until we've updated analyzer_cli's pubspec to point to at least analyzer version 0.25.2-alpha.1, which is the first version of analyzer that has null-aware operators enabled by default.  Finally, when we do make this change, I think we should a deprecation message to let the user know that the "enable null aware operators" flag is no longer needed (see #23899).